Source/WebCore/platform/ScrollSnapAnimatorState.cpp

```diff
@@ -141,21 +141,37 @@ HashSet<ElementIdentifier> ScrollSnapAnimatorState::currentlySnappedBoxes(const
     return snappedBoxIDs;
 }
 
-ElementIdentifier ScrollSnapAnimatorState::chooseBoxToResnapTo(const Vector<SnapOffset<LayoutUnit>>& horizontalOffsets, const Vector<SnapOffset<LayoutUnit>>& verticalOffsets) const
+void ScrollSnapAnimatorState::setActiveSnapIndexForAxis(ScrollEventAxis axis, std::optional<unsigned> index)
 {
-    auto found = std::find_if(horizontalOffsets.begin(), horizontalOffsets.end(), [this](SnapOffset<LayoutUnit> p) -> bool {
-        return m_currentlySnappedBoxes.contains(p.snapTargetID) && p.isFocused;
+    setActiveSnapIndexForAxisInternal(axis, index);
+    updateCurrentlySnappedBoxes();
+}
+
+void ScrollSnapAnimatorState::updateCurrentlySnappedBoxes()
+{
+    auto horizontalOffsets = currentlySnappedOffsetsForAxis(ScrollEventAxis::Horizontal);
+    auto verticalOffsets = currentlySnappedOffsetsForAxis(ScrollEventAxis::Vertical);
+
+    m_currentlySnappedBoxes = currentlySnappedBoxes(horizontalOffsets, verticalOffsets);
+}
+
+static ElementIdentifier chooseBoxToResnapTo(const HashSet<ElementIdentifier>& snappedBoxes, const Vector<SnapOffset<LayoutUnit>>& horizontalOffsets, const Vector<SnapOffset<LayoutUnit>>& verticalOffsets)
```

Comment:
I’m a bit confused why this was made a static function. I think it only makes sense to run chooseBoxToResnapTo() on m_currentlySnappedBoxes right?

Reply:
I wanted to make the update step of `m_currentlySnappedBoxes` very explicit, hence the “move out of the member var, and pass to this static function” approach.